src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -827,17 +837,26 @@ private void ImportCall(ILOpcode opcode, int token)
             {
                 throw new NotImplementedException();
             }
-            if (opcode == ILOpcode.callvirt && callee.IsAbstract)
-            {
-                throw new NotImplementedException();
-            }
 
             if (callee.OwningType.IsDelegate)
             {
-                throw new NotImplementedException();
+                if (opcode == ILOpcode.newobj)
+                {
+                    FunctionPointerEntry functionPointer = ((FunctionPointerEntry)_stack.Peek());
+                    DelegateCreationInfo delegateInfo = _compilation.GetDelegateCtor(callee.OwningType, functionPointer.Method, functionPointer.IsVirtual);
+                    callee = delegateInfo.Constructor.Method;
+                    if(callee.Signature.Length == 3)
+                    {
+                        PushExpression(StackValueKind.NativeInt, "thunk", GetOrCreateLLVMFunction(_compilation.NodeFactory.NameMangler.GetMangledMethodName(delegateInfo.Thunk.Method).ToString()));
+                    }
+                }
+                else
+                {
+                    opcode = ILOpcode.call;
```

Comment:
Why do we need to bash callvirt to call on delegates?

(If this is to optimize `Invoke`, the way `Invoke` is optimized in other codegens is by loading a `this` from a known offset, loading a function address from a known address, and calling that address - we only very rarely execute the generated body of the `Invoke` method).

Reply:
Removed the bashing